lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1693,6 +1697,12 @@ def plot_surface(self, X, Y, Z, *args, norm=None, vmin=None,
                         cbook._array_perimeter(a[rs:rs_next+1, cs:cs_next+1])
                         for a in (X, Y, Z)
                     ]
+                    # If any of the perimeters are masked, then skip the
```

Comment:
Can you test this?

Reply:
I've added another test for this based on the example [above](https://github.com/matplotlib/matplotlib/pull/18114#issuecomment-876292099)

Reply:
Is this behavior desirable? If everything _but_ the perimeter of the polygon is masked, do we really want to draw the polygon? Some other choices would be:

* Draw the polygon only if _none_ of the contained elements are masked
* Draw the polygon if less than half (or a user-configurable parameter of the contained elements are masked
* Set the opacity of the polygon to be proportional to the maskedness
* Attempt to precisely outline the masked regions

Whatever you decide on, I think it's important to describe the choice in the docstring.

Reply:
It is a good question and I'm happy to have more input on this topic.

My line of thought is that as long as the perimeters of the polygons don't contain masked data, then it is ok to draw the polygon. If some data inside of the polygon is masked but is strided over, then it is not used to make the plot anyway so we aren't "revealing" masked data in the plot. I understand this might cause a bit of confusion due to the nature of the `xstride/xcount` settings.

> Draw the polygon if less than half (or a user-configurable parameter of the contained elements are masked

and

> Set the opacity of the polygon to be proportional to the maskedness

These seem like "advanced" use cases to me but happy to hear if others think differently. Though I still don't think anyone would want the perimeter of a polygon to contain masked data. As far as I can tell, no one has requested this kind of feature before.

> Attempt to precisely outline the masked regions

Probably the ideal solution, if any one has suggestions how the can be accomplished then I'm happy to hear them!

Reply:
I think I want input from @ianthomas23 about how he handled masked data for the contouring functions. I think it would make sense to be consistent with those rules.

Reply:
Of course, the concepts don't perfectly overlap because the contouring algorithms don't have a concept of striding, but I am still interested in his input.

Reply:
> I don't believe striding does anything other than a simple decimation at the moment?

It doesn't decimate the space fully, as the perimeter is still drawn at full precision. But looking closely, the colormap color is computed based only on the perimeter, so I guess the implemented behavior is consistent.